ci/konflux/generate_component_definitions.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+import yaml
+
+
+"""
+This script is used to configure a Konflux Application with component definitions.
+We have very many components, and clicking them one by one in the UI is too inefficient.
+
+$ poetry run ci/cached-builds/konflux_generate_component_definitions.py > konflux_components.yaml
+$ oc apply -f konflux_components.yaml
+
+Open https://console.redhat.com/application-pipeline/workspaces/rhoai-ide-konflux/applications
+and see the result in the "Components" tab.
+"""
+
+workspace_name = "rhoai-ide-konflux-tenant"
+application_name = "kubeflow"
+application_uid = "89959242-a304-41ef-9654-c360c415fbb9"
+git_revision = "main"
+git_url = "https://github.com/opendatahub-io/kubeflow"
+pr_number = "514"
```

Comment:
is this for testing purpose ?

Reply:
No, konflux tracks what was the pull-request that introduced the tekton pipelines into the project repo. It's visible in the UI here (the `Merge in Github` button) and when I don't set this, things behave strangely

![image](https://github.com/user-attachments/assets/1c55fafd-d967-46cb-b870-bec98291328f)

https://console.redhat.com/application-pipeline/workspaces/rhoai-ide-konflux/applications/kubeflow/components

Reply:
thanks for the information.